### PR TITLE
More consistent Ember lint rules

### PR DESCRIPTION
**Pull request by arjansingh**
> What changed? 

Added rules around preferring `const` and multi-line arrays and un-nested `if` statements.

Made some rules warnings instead of ignored.

> Are there gotchas?

How should we bump the versioning on this?

---

<details>

<summary>Pull Request Todo</summary>

- [ ] Solicit reviews from domain experts or others outside this PR's squad

</details>

<details>

<summary>Post-Deploy Todo</summary>

- [ ] Monitor exceptions in Sentry – https://goo.gl/gB1t9Z
- [ ] Monitor key metrics like pixel fires and asset sizes – https://goo.gl/BkVpRg
- [ ] Spot-check production – http://dollarshaveclub.com

_In the event an outage related to this PR occurs, rollback immediately, then debug: https://jenkins-deploy.aws.shave.io/job/Production/job/rollback/_

</details>

<details>

<summary>Code Review Guidelines</summary>

- Ensure readability and good organization
- Ensure classes, functions, components, and variables are well-named
- Ensure the diff is small
- Ensure this PR does only one thing
- Ensure the code is DRY (more than twice repeated)
- Ensure tricky code is removed or explained with a comment
- Ensure `TODO` comments and `WHY` comments are added where applicable

</details>

### CODE REVIEW

File: ember.js [hunk 1]
```diff
@@ -12,11 +12,14 @@ module.exports = {
       '_super',
       '__container__',
     ]}],
-    'consistent-return': 0,
+    'consistent-return': 1,
```
Comment (jonathanong) [inline ember.js:15]: I find this rule annoying because it dislikes early returns

**Comment by jonathanong** [comment]
Otherwise 👍

**Review by briangonzalez — CHANGES_REQUESTED** [review]
Tests please.

File: ember.js (hunk 1 above)
Comment (arjansingh) [reply ember.js:15]: That's why it's a warning. I think it's fine to leave it up to developer discretion.

https://eslint.org/docs/user-guide/configuring#configuring-rules

```
"off" or 0 - turn the rule off
"warn" or 1 - turn the rule on as a warning (doesn’t affect exit code)
"error" or 2 - turn the rule on as an error (exit code is 1 when triggered)
```

**Comment by arjansingh** [comment]
Good call @briangonzalez

**Review by briangonzalez — APPROVED** [review]

**Pull request merged**